### ISSUE TITLE
[CST][ENG] Update File Uploader to use the new `va-file-input-multiple` component

### ISSUE THREAD

**Issue by pmclaren19** [department-of-veterans-affairs/va.gov-team#87835]
## Background Context
Initially created a bug ticket [here](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2549) for the Platform Design team to update the va-file-input component to allow for multiple files. They took this ticket and created three new tickets to cover this work:

- va-icon-button - #[2641](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2641) (needed for file input)
- update file input - #[2642](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2642) (needs to happen before multiple file input)
- multiple file input - #[2643](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2643)

This work has now been completed and we can update the file upload component that we are using in the CST on the Doc Request Page and Files Tab to use this new component [va-file-input-multiple](https://design.va.gov/storybook/?path=/docs/uswds-va-file-input-multiple--docs#additional-form-inputs) 

### New Feature
Update the AddFilesForm component that allows users to upload files to now instead us the component [va-file-input-multiple](https://design.va.gov/storybook/?path=/docs/uswds-va-file-input-multiple--docs#additional-form-inputs) . This component is used on the Files Tab and Document Request Form

## Design

**Current:**
<img width="794" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBelhBQlE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--870c94b0661c31b573ee17b50be95e6af6434d31/Screenshot%202024-07-08%20at%202.23.27%E2%80%AFPM.png" alt="Screenshot 2024-07-08 at 2.23.27 PM.png" />

**va-file-input-multiple component:**
<img width="603" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBemZBQlE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--3e2b18d0579a742e5ca2fb25bcd0b12633fbf4a7/Screenshot%202024-07-08%20at%202.03.43%E2%80%AFPM.png" alt="Screenshot 2024-07-08 at 2.03.43 PM.png" />

- @skylerschain is a design mock needed?

### Enablement team (if needed)
@davidakennedy ?

## Engineering 
- We will want to update the component `AddFileForms` to use https://design.va.gov/storybook/?path=/docs/uswds-va-file-input-multiple--docs#additional-form-inputs

## Out of scope
- Updating the AddFilesFormOld component that is only enabled when the feature flag cst_use_claim_details_v2 is disabled

## Open questions
- Do we need to use a feature flag?
- Do we need to update the ask va page as well?

## Tasks
- [ ] The files tab now uses the new va component to upload multiple files
- [ ] The Doc Request page now uses the new va component to upload multiple files
- [ ] Users see a drag and drop instead of a button for adding files
- [ ] Users can add multiple files 
- [ ] When a user adds an encrypted file they are able to populate the password in an input field for that file
- [ ] When a user adds a file they are able to select from a drop downt he doc type for that file
- [ ] unit tests are added/updated
- [ ] cypress tests are added/updated

## Definition of Done
- [ ] Meets acceptance criteria
- [ ] Passed E2E testing (90% coverage)
- [ ] Passed unit testing (90% coverage)
- [ ] Passed integration testing (if applicable)
- [ ] Code reviewed (internal)
- [ ] Submitted to staging
- [ ] Team approved production verification process
- [ ] Design performs design QA and verifies the implementation matches the design spec
- [ ] Accessibility specialist performs accessibility review (in code or design)
- [ ] Engineering identifies staging users required to test and shares account and credentials with design and product
- [ ] Product performs functional QA and verifies acceptance criteria was met

## Acceptance Criteria
- [ ] The files tab now uses the new va component to upload multiple files
- [ ] The Doc Request page now uses the new va component to upload multiple files
- [ ] Users see a drag and drop instead of a button for adding files
- [ ] Users can add multiple files 
- [ ] When a user adds an encrypted file they are able to populate the password in an input field for that file
- [ ] When a user adds a file they are able to select from a drop downt he doc type for that file

**Comment by pmclaren19** [comment]
Reviewed and added some updates to tasks during eng refinement.

**Comment by pmclaren19** [comment]
Running into some issues implementing this code. Talking with platform design and created several bug tickets ...
- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3139

**Comment by pmclaren19** [comment]
This ticket is blocked by the following platform design tickets:

- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3139
- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3153

**Comment by jacobworrell** [comment]
Still awaiting updates from Platform.

**Comment by skylerschain** [comment]
It looks the bugs Peri filed and linked to above are Closed. So is this ticket unblocked now? I added the latest design mock to the Design section.

**Comment by pmclaren19** [comment]
slack comments- https://dsva.slack.com/archives/C01DBGX4P45/p1722966284369869
Both bug tickets have been worked. We need to attempt to implement again and see if the slot can be conditional.

**Comment by pmclaren19** [comment]
> I also want to highlight a bug I noticed while looking at this in staging:
> 
> Repro:
> 
> 1. User uploads a file (File 1)
> 2. Gray "attached" box (which we will be removing as part of this ticket) shows up, confirming that File 1 has been uploaded
> 3. User clicks "Change file" to replace File 1
> 4. User replaces File 1 with File 2
> 5. The gray "attached" boxes show up for both File 1 and File 2, even though File 1 is no longer attached.
> 
> I assume this would get fixed when we remove that gray box and replace it with the pattern from the design system. But wanted to document this just to be safe.
> 
>  Screen.Recording.2024-10-25.at.10.55.00.AM.mov

@skylerschain for the time being while we try to fix this please file a separate ticket to fix the issue witht he current implementation.

**Comment by pmclaren19** [comment]
Working this ticket and found several bugs with the `va-file-input-multiple` component. Talked to the platform design team and we wrote up these tickets. Likely they wont be worked until Jan 2025. 

- [Ticket for vaMultipleChange firing twice](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3549)
- [Ticket for disallowing multiple of the same file](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3550)
- [Conditional slot content ticket](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3551)

**Comment by pmclaren19** [comment]
Talked with platform design and we created another bug ticket for [Update va-file-input-multiple's remove file emitted event ](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3597)

**Comment by pmclaren19** [comment]
Talked with platform design and we created another bug ticket for [Encrypted file ticket](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3601)

**Comment by pmclaren19** [comment]
Hey Team talked to the platform design team this morning about the va-file-input-multiple component and sound like they are going to start working on those changes this upcoming sprint: Sprint 19 (Jan 16 to 29). :thread:  [here](https://dsva.slack.com/archives/C01DBGX4P45/p1736175433305579).

**Comment by pmclaren19** [comment]
Update:

Looks like the following have been completed:
[Update va-file-input-multiple's remove file emitted event vets-design-system-documentation](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3597)
[va-file-input-multiple emits twice on file upload](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3549va-file-input-multiple)
[va-file-input-multiple allow for showing the slot conditionally](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3551)
We are still waiting on these other two to be worked and completed:
[Encrypted file ticket](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3601)
[va-file-input-multiple allows the same file to be uploaded more than once vets-design-system-documentation#3550](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3550)

**Comment by pmclaren19** [comment]
Attempted to implement and ran into issues. Created this bug ticket: https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3785

**Comment by MikeC-A6** [comment]
@janechodance Amy mentioned that there might have been some handoff from the previous BMT1 team for this file upload component work. From what I can tell, it looks like the issue that was referenced in the comment above was opened with the platform team, and they prioritized that work for this sprint. Do you have any additional insight on this that is not part of this thread or in [#3785?](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3785) ? CC @amylai-va
